src/client.zig

```diff
@@ -21,9 +21,20 @@ const CommitTransfersResult = tb.CommitTransfersResult;
 
 const log = std.log;
 
+pub const ClientError = packed enum(u32) {
+    max_message_size_exceeded,
+    batch_not_queued,
+};
+
+const empty_result: [0]u8 = .{};
 pub const Client = struct {
     const Request = struct {
-        const Callback = fn (user_data: u128, operation: Operation, results: []const u8) void;
+        const Callback = fn (
+            user_data: u128,
+            operation: Operation,
+            results: []const u8,
+            client_error: ?u32,
```

Comment:
Great idea to pass the error to the callback instead of bifurcating the return type of the `request` method, which is what I was assuming. Your approach is more flexible and future proof, and easier to handle for the caller in a single place.

We can actually combine `client_error` into `results` by making `results` a Zig error union, e.g. `ClientError![]const u8`, and by converting `ClientError` to be a Zig error set.

The benefit of this is that we can get rid of the `client_error` arg for less surface area, and also the `empty_result` global and it also makes it impossible for someone to ignore the error, whereas at present, they could work with `empty_result` and not check the error and get into trouble.

Here are some examples of returning an error or result when calling a callback that we did in IO: https://github.com/coilhq/tigerbeetle/blob/main/src/io.zig#L429